### PR TITLE
Stripe checkout customization cookbook

### PR DESCRIPTION
**Pull request by bruno-ds**
| Q               | A
| --------------- | -----
| Branch?         | 1.2
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Slack discussion | https://sylius-devs.slack.com/archives/C3E4EBKUZ/p1533625371000023?thread_ts=1533559529.000181&cid=C3E4EBKUZ
| License         | MIT

Hello as discussed in slack please find a cookbook proposal.
It leverage the possibility to : 
- override Payum layout & Payum Stripe templates
- Extend Payum Stripe  gateway in order to add metadata

Those changes are in qualification in our integration servers and will be deployed in the next release of https://store.itophub.io

### CODE REVIEW

**Comment by bruno-ds** [comment]
Friendly ping to @CoderMaggie 🙊

**Comment by bruno-ds** [comment]
I should be punished for pigging on weekends only, but I have no time during the week
(Ping🎾)

File: docs/cookbook/payments/stripe-customization.rst [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+        <script type="text/javascript">
+            $(function() {$('form .stripe-button-el').click();});
+        </script>
+    {% endblock payum_javascripts %}
+
+Add metadata to Stripe
+---------------------------------------------------------------------------
+
+The metadata have to be given later to stripe, so you can't simply use the template overriding, instead, you have to define a Payum Extension that will hanbdle this for you at the right time:
```
Comment (pamil) [inline docs/cookbook/payments/stripe-customization.rst]: `hanbdle` -> `handle`

File: docs/cookbook/payments/stripe-customization.rst [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            $payment = $request->getModel();
+
+            /** @var Order $order */
+            $order = $payment->getOrder();
+
+            /** @var Customer $customer */
+            $customer = $order->getCustomer();
+
+            $context->getGateway()->execute($status = new GetStatus($payment));
+            if (! $status->isNew()) {
```
Comment (pamil) [inline docs/cookbook/payments/stripe-customization.rst]: `! $status->isNew()` -> `!$status->isNew()` to comply with Sylius CS

File: docs/cookbook/payments/stripe-customization.rst [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+    # app/config/config.yml
+
+    app.payment.payum.stripe.on_capture.add_metadata:
+        class: AppBundle\Payment\StripeAddMetadataOnCaptureExtensions
+        tags:
+            - { name: payum.extension, factory: stripe_checkout, prepend: true}
+
+
+From now on, your service will be acalled each time a Stripe's Catpure is performed and it will add metadata to the payment, they will be visible on stripe's side.
```
Comment (pamil) [inline docs/cookbook/payments/stripe-customization.rst]: `acalled` -> `called`

**Comment by pamil** [comment]
I've rebased the PR to the 1.2 branch. Sorry for the delayed response!

**Comment by bruno-ds** [comment]
Sorry for the delay, Sphinx could not be installed on my setup, so I had to delay my corrections.
Since I have to rebase and so much things (and I am not an expert of git) I close this PR and open a new one on the 1.2 branch.

**Pull request closed without merging**